### ISSUE TITLE
runBatch is letting through non-numeric results (?)

### ISSUE THREAD

**Issue by jeanlucj** [jeanlucj/BreedSimCost#1]
I'm not sure if that's what it was.  Here's the backtrace.  The optimization parameters are below that.
Backtrace:
     █
  1. ├─rmarkdown::render("analysis/BreedSimCostTest.Rmd")
  2. │ └─knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
  3. │   └─knitr:::process_file(text, output)
  4. │     ├─base::withCallingHandlers(...)
  5. │     ├─knitr:::process_group(group)
  6. │     └─knitr:::process_group.block(group)
  7. │       └─knitr:::call_block(x)
  8. │         └─knitr:::block_exec(params)
  9. │           └─knitr:::eng_r(options)
 10. │             ├─knitr:::in_dir(...)
 11. │             └─knitr:::evaluate(...)
 12. │               └─evaluate::evaluate(...)
 13. │                 └─evaluate:::evaluate_call(...)
 14. │                   ├─evaluate:::timing_fn(...)
 15. │                   ├─base:::handle(...)
 16. │                   ├─base::withCallingHandlers(...)
 17. │                   ├─base::withVisible(eval(expr, envir, enclos))
 18. │                   └─base::eval(expr, envir, enclos)
 19. │                     └─base::eval(expr, envir, enclos)
 20. ├─BreedSimCost::optimizeByLOESS(bsd)
 21. │ └─BreedSimCost::runBatch(batch, bsd)
 22. │   └─batchResults %>% mutate(response = endVarMean - initPopMean)
 23. ├─dplyr::mutate(., response = endVarMean - initPopMean)
 24. ├─dplyr:::mutate.data.frame(., response = endVarMean - initPopMean)
 25. │ └─dplyr:::mutate_cols(.data, ..., caller_env = caller_env())
 26. │   ├─base::withCallingHandlers(...)
 27. │   └─mask$eval_all_mutate(quo)
 28. └─base::.handleSimpleError(...)
 29.   └─dplyr:::h(simpleError(msg, call))

Optimization parameters:
# Implementation details for optimization
# Get rid of parallel processing if debugging
debug
FALSE

# Print stuff as the optimization goes along
verbose
FALSE

# Save intermediate results as you go along
saveIntermediateResults
TRUE

# Number of cores for parallel processing
nCores
4

# parameters for initial grid search
# each percentage vector should be nStages + 1
# The first value is for the population improvement cycle
minPercentage
0.10 0.20 0.05 0.05

maxPercentage
0.70 0.80 0.45 0.45

percentageStep
0.20 0.20 0.20 0.20

# Minimum number of breeding progeny in PIC
minNBreedingProg
2

# How small should the confidence interval
# on budgets be before you stop
tolerance
0.08

# When running batches of simulations, how many sims per batch?
batchSize
36

# How many batches should you do before you quit
maxNumBatches
20

# When redoing simulations, how many should be high gain?
nHighGain
18

#When redoing simulations, how many should be high uncertainty?
nUncertain
12

**Comment by jeanlucj** [comment]
The problem is that the budget being tested for optimality may need more entries in the various variety candidate stages than were used during the phenotypic selection burnin period

**Comment by jeanlucj** [comment]
Resolved by a workaround:
If there are too few entries when the two-part strategy starts, chooseTrialEntries calls makeVarietyCandidates to make more, enough to fill the pipeline.
This will generally happen at Stage 2 because Stage 1 get filled itself by makeVarietyCandidates.
It's tricky to figure out from what generation of the breedingPop the _parents_ for makeVarietyCandidates should come from.  I hope I figured that out correctly.
